tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/AIRedactionalHelper.java

```diff
@@ -53,15 +53,14 @@ public Mono<String> suggestContent(String userInput, Optional<String> mailConten
 
         List<ChatMessage> messages = Stream.concat(
             Stream.of(promptForContext, promptForUserInput),
-            mailContent.map(content -> Stream.of(new SystemMessage(content))).orElseGet(Stream::empty)
-        ).collect(Collectors.toUnmodifiableList());
+            mailContent.stream().map(SystemMessage::new))
+            .collect(Collectors.toUnmodifiableList());
```

Comment:
```suggestion
            .collect(ImmutableList.toImmutableList());
```

Immutable != un-modifiable